src/mpi/coll/include/coll_impl.h

```diff
@@ -105,4 +105,27 @@ void MPIR_Coll_host_buffer_alloc(const void *sendbuf, const void *recvbuf, MPI_A
                                  MPI_Datatype datatype, void **host_sendbuf, void **host_recvbuf);
 void MPIR_Coll_host_buffer_free(void *host_sendbuf, void *host_recvbuf);
 
+#define MPII_COLL_HOST_BUFFER_COPY(host_sendbuf_, host_recvbuf_, in_recvbuf_, count_, datatype_, request_) \
```

Comment:
I ran out of time to work on this more today, but here is an initial draft of the macro.